homeassistant/components/soma/cover.py

```diff
@@ -67,3 +67,8 @@ def current_cover_position(self):
     def is_closed(self):
         """Return if the cover is closed."""
         return self.current_position == 0
+
+    @property
+    def device_state_attributes(self):
+        """Get additional state attributes."""
+        return self.state_attrs
```

Comment:
Battery level should be reported as a separate sensor entity.

Reply:
This is different from smartthings/cover.py then, that I was using as a guide? I'm not overly keen on having sensor.<blindname> for just battery level really, especially when it is directly an attribute of the blind/cover motor

Reply:
We have quite a bit of legacy around, from an older era. Since we now have the notion of devices that have entities, we can expose more as separate entities. Entities are easier and more friendly to use in automations and in the UI for the end-user.

Hence the request to make it a separate entity.

Reply:
Cool, will sort it out.